bgpd/bgp_nht.c

```diff
@@ -416,6 +416,20 @@ void bgp_parse_nexthop_update(int command, vrf_id_t vrf_id)
 
 			nexthop = nexthop_from_zapi_nexthop(&nhr.nexthops[i]);
 
+			/*
+			 * Turn on RA for the v6 nexthops
+			 * we receive from bgp.  This is to allow us
+			 * to work with v4 routing over v6 nexthops
+			 */
+			if (nhr.prefix.family == AF_INET6) {
+				struct interface *ifp;
+
+				ifp = if_lookup_by_index(nexthop->ifindex,
```

Comment:
Should we have a NULL check for ifp ?

Being it an interface structure there is always a chance of interface getting deleted and ifp can be NULL. I see that zclient_send_interface_radv_req simply uses ifp and there is null check in this function.

Reply:
I'm not worried about this actually happening.  If we are using zebra then we will have up-to-date information about interfaces.  If we crash here that means we have a serious bug that needs to be resolved between zebra and bgp communications.

Reply:
I checked the code around and I think I understood your point. As BGP just received the the update message from Zebra, and as BGP is processing this message, there is no way the interface get deleted and ifp being NULL in between. Is that right ? I am little skeptical as these are two different processes. Let me know, I can approve accordingly.

Reply:
yes that is impossible.